crates/typst/src/export/pdf/color.rs

```diff
@@ -0,0 +1,382 @@
+const HSV_S: Name<'static> = Name(b"HsvS");
+
+/// The name of the "value" component of the HSV color space
+const HSV_V: Name<'static> = Name(b"HsvV");
+
+/// The name of the "hue" component of the HSL color space
+const HSL_H: Name<'static> = Name(b"HsvH");
+
+/// The name of the "saturation" component of the HSL color space
+const HSL_S: Name<'static> = Name(b"HsvS");
+
+/// The name of the "lightness" component of the HSL color space
+const HSL_L: Name<'static> = Name(b"HsvL");
+
+/// Deflated sRGB ICC profile
+#[comemo::memoize]
+fn srgb_icc() -> Arc<Vec<u8>> {
+    Arc::new(deflate(SRGB_ICC))
+}
+
+/// Deflated gray ICC profile
+#[comemo::memoize]
+fn gray_icc() -> Arc<Vec<u8>> {
+    Arc::new(deflate(GRAY_ICC))
+}
+
+/// Deflated Oklab PostScript function
+#[comemo::memoize]
+fn oklab_function() -> Arc<Vec<u8>> {
+    let code = minify(OKLAB_SOURCE);
+    Arc::new(deflate(code.as_bytes()))
+}
+
+/// Deflated HSV PostScript function
+#[comemo::memoize]
+fn hsv_function() -> Arc<Vec<u8>> {
+    let code = minify(HSV_SOURCE);
+    Arc::new(deflate(code.as_bytes()))
+}
+
+/// Deflated HSL PostScript function
+#[comemo::memoize]
+fn hsl_function() -> Arc<Vec<u8>> {
+    let code = minify(HSL_SOURCE);
+    Arc::new(deflate(code.as_bytes()))
+}
+
+/// This function removes comments, line spaces and carriage returns from a
+/// PostScript program. This is necessary to optimize the size of the PDF file.
+fn minify(source: &str) -> String {
+    let mut buf = String::with_capacity(source.len());
+    let mut in_comment = false;
+    let mut in_line_space = false;
+    for c in source.chars() {
+        match c {
+            '%' => in_comment = true,
+            '\n' => {
+                if !in_line_space {
+                    buf.push(' ');
+                }
+
+                in_comment = false;
+                in_line_space = true;
+            }
+            '\r' => {}
+            _ if in_comment => {}
+            _ if in_line_space => {
+                if !c.is_whitespace() {
+                    in_line_space = false;
+                    buf.push(c);
+                }
+            }
+            _ => {
+                buf.push(c);
+            }
+        }
+    }
+    buf
+}
+
+pub trait ColorPdfEncode {
+    /// Encodes the color into four f32s, which can be used in a PDF file.
+    /// Ensures that the values are in the range [0.0, 1.0].
+    ///
+    /// # Why?
+    /// - Oklab: The a and b components are in the range [-0.4, 0.4] and the PDF
+    ///   specifies (and some readers enforce) that all color values be in the range
+    ///   [0.0, 1.0]. This means that the PostScript function and the encoded color
+    ///   must be offset by 0.4.
+    /// - HSV/HSL: The hue component is in the range [0.0, 360.0] and the PDF format
+    ///   specifies that it must be in the range [0.0, 1.0]. This means that the
+    ///   PostScript function and the encoded color must be divided by 360.0.
```

Comment:
I think the longer-form documentation should be on the trait and the method itself can just a brief oneliner.

Reply:
On second thought, why is this a trait? I think, a free-standing function would be simpler here.

Reply:
it's a pattern done throughout the code base of having small traits of "local" functions on types such as `AbsExt`
![image](https://github.com/typst/typst/assets/9665250/bdd7b860-81d5-4815-89ff-f74d8437e76f)

Reply:
In the meantime, I'll apply your initial suggestion ;)